openmc/statepoint.py

```diff
@@ -606,12 +622,17 @@ def link_with_summary(self, summary):
 
         Raises
         ------
+        RuntimeError
+            If a Summary object has already been linked.
         ValueError
             An error when the argument passed to the 'summary' parameter is not
             an openmc.Summary object.
 
         """
 
+        if self.summary is not None:
+            raise RuntimeError('A Summary object has already been linked.')
```

Comment:
Perhaps we might raise a warning here instead of an error? It's not a big deal, but I don't see a reason to necessarily error out here. Certainly not a major issue though.